core/livepeernode.go

```diff
@@ -49,10 +50,16 @@ type LivepeerNode struct {
 	SegmentChans  map[int64]SegmentChan
 	Ipfs          ipfs.IpfsApi
 	ServiceURI    *url.URL
+	OrchSecret    string
+	Transcoder    Transcoder
 
 	// Transcoder private fields
 	claimMutex   *sync.Mutex
 	segmentMutex *sync.Mutex
+	tcoderMutex  *sync.Mutex
+	taskMutex    *sync.Mutex
+	taskChans    map[int64]TranscoderChan
+	taskCount    int64
```

Comment:
More of a comment here - the `LivepeerNode` is getting pretty big and hard to reason about.  I hope breaking it into `LivepeerBroadcastNode`, `LivepeerOrchestratorNode` and `LivepeerTranscoderNode` can help separate the responsibilities.

Reply:
Sure, perhaps as a quick follow-on after this is merged? Also want to ensure that it doesn't make merging the other WIP PRs too much of a headache.

We could name these `BroadcastNode` , `OrchestratorNode` and so forth since the "Livepeer" part is implicit, to save on the verbosity?

Reply:
Yeah that makes sense